Algorithm.Framework/Portfolio/BlackLittermanOptimizationPortfolioConstructionModel.py

```diff
@@ -105,8 +115,10 @@ def CreateTargets(self, algorithm, insights):
                 # the new combined expected return vector
                 expected_return = dot(inv(A), B)
 
-        # the optimization method processes the data frame
-        opt, weights = self.maximum_sharpe_ratio(self.risk_free_rate, expected_return, returns)
+        # The portfolio optimizer finds the optional weights for the given data
+        self.optimizer.expected_returns = expected_return
```

Comment:
So, you assume that any optimizer would have `expected_returns` field.

Reply:
Good point. In `__init__` we will have to check whether the optimizer has a `expected_returns` attribute and raise an exception if not.